File: test/demo/wslc-polyglot-hello/polyglot/servers/srv.ml
let () =
  let port = int_of_string Sys.argv.(1) in
  let s = Unix.socket Unix.PF_INET Unix.SOCK_STREAM 0 in
  Unix.setsockopt s Unix.SO_REUSEADDR true;
  Unix.bind s (Unix.ADDR_INET (Unix.inet_addr_any, port));
  Unix.listen s 1;
  let (c,_) = Unix.accept s in
  let b = Bytes.create 64 in
  ignore (Unix.read c b 0 64);
  let m = "hello world from [ocaml]\n" in
  ignore (Unix.write_substring c m 0 (String.length m));
  ignore (Unix.read c b 0 64);
  Unix.close c

File: test/demo/wslc-polyglot-hello/polyglot/servers/srv.py
import socket
import sys

port = int(sys.argv[1])

s = socket.socket()
s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
s.bind(('0.0.0.0', port))
s.listen(1)

conn, _ = s.accept()
conn.recv(64)                                  # Windows request
conn.sendall(b'hello world from [python]\n')
conn.recv(64)                                  # Windows ack -> exit
conn.close()
